File: src/main/java/com/liamfarrell/android/snapbattle/activity/createbattle/ChooseVotingFragment.java
package com.liamfarrell.android.snapbattle.activity.createbattle;

import android.content.Context;
import android.os.Bundle;
import android.support.v4.app.Fragment;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.AdapterView;
import android.widget.ArrayAdapter;
import android.widget.CheckedTextView;
import android.widget.RadioButton;
import android.widget.RadioGroup;
import android.widget.Spinner;

import com.liamfarrell.android.snapbattle.app.App;
import com.liamfarrell.android.snapbattle.R;

import java.util.ArrayList;

/**
 * Created by liamf on 28/02/2018.
 */

public class ChooseVotingFragment extends Fragment implements View.OnClickListener  {

    private Spinner mVotingLengthSpinner;
    private VotingChoice mChosenVotingChoice;



    @Override
    public void onClick(View view) {
        // determine which voting option is checked and update the member variables
        boolean checked = ((RadioButton) view).isChecked();

        // Check which radio button was clicked
        switch (view.getId()) {
            case R.id.noVotingButton:
                if (checked) {
                    mVotingLengthSpinner.setEnabled(false);
                    mChosenVotingChoice = VotingChoice.NONE;
                }
                break;
            case R.id.publicVotingButton:
                if (checked) {
                    mVotingLengthSpinner.setEnabled(true);
                    mChosenVotingChoice = VotingChoice.PUBLIC;
                }
                break;


            case R.id.mutualFacebookFriendsButton:
                if (checked)
                {
                    mVotingLengthSpinner.setEnabled(true);
                    mChosenVotingChoice = VotingChoice.MUTUAL_FACEBOOK;
                }

                break;
        }
    }


    public enum VotingChoice {
        MUTUAL_FACEBOOK {
            public String toString() {
                return App.getContext().getResources().getText(R.string.mutual_facebook_friends).toString();
            }
            public String getLongStyle(){
                return App.getContext().getResources().getText(R.string.mutual_facebook_friends_long).toString();
            }
        },
        PUBLIC {
            public String toString() {
                return App.getContext().getResources().getText(R.string.public_voting).toString();
            }
            public String getLongStyle(){
                return App.getContext().getResources().getText(R.string.public_voting_long).toString();
            }
        },
        SELECTED {
            public String toString() {
                return App.getContext().getResources().getText(R.string.selected_judges).toString();
            }
            public String getLongStyle(){
                return App.getContext().getResources().getText(R.string.selected_judges).toString();
            }
        },
        NONE {
            public String toString() {
                return App.getContext().getResources().getText(R.string.no_voting_used).toString();
            }
            public String getLongStyle(){
                return App.getContext().getResources().getText(R.string.no_voting_used).toString();
            }
        };
        public abstract String getLongStyle();
    }


    public enum VotingLength {
        TWENTY_FOUR_HOURS {
            public String toString(Context context) {
                return context.getResources().getStringArray(R.array.VotingLengths)[0];
            }
        },
        THREE_DAYS {
            public String toString(Context context) {
                return context.getResources().getStringArray(R.array.VotingLengths)[1];
            }
        },
        ONE_WEEK {
            public String toString(Context context) {
                return context.getResources().getStringArray(R.array.VotingLengths)[2];
            }
        },
        ONE_MONTH {
            public String toString(Context context) {
                return context.getResources().getStringArray(R.array.VotingLengths)[3];
            }
        };
        public abstract String toString(Context context);
    }

    @Override
    public View onCreateView(LayoutInflater inflater, ViewGroup container, Bundle savedInstanceBundle) {


        View view = inflater.inflate(R.layout.fragment_choose_voting, container, false);


        mVotingLengthSpinner = view.findViewById(R.id.votingLengthSpinner);
        RadioGroup votingTypeRadioGroup = view.findViewById(R.id.votingTypeRadioGroup);

        view.findViewById(R.id.noVotingButton).setOnClickListener(this);
        view.findViewById(R.id.publicVotingButton).setOnClickListener(this);
        view.findViewById(R.id.mutualFacebookFriendsButton).setOnClickListener(this);


        votingTypeRadioGroup.check(R.id.noVotingButton);
        mChosenVotingChoice = VotingChoice.NONE;


        ArrayAdapter<VotingLength> adapter = new VotingLengthAdapter(getActivity());


        // Specify the layout to use when the list of voting length choices appears
        //adapter.setDropDownViewResource(android.R.layout.simple_spinner_dropdown_item);

        // Apply the adapter to the spinner
        mVotingLengthSpinner.setAdapter(adapter);
        mVotingLengthSpinner.setEnabled(false);
        mVotingLengthSpinner.setSelection(0);


        return view;
    }



    public VotingChoice getChosenVotingChoice()
    {
        return mChosenVotingChoice;
    }
    public VotingLength getChosenVotingLength()
    {
        return (VotingLength)mVotingLengthSpinner.getSelectedItem();
    }





        public class VotingLengthAdapter extends ArrayAdapter<VotingLength> {
        private Context con;
        public VotingLengthAdapter (Context context) {
            super(context, 0, VotingLength.values());
            con = context;
        }

        @Override
        public View getView(int position, View convertView, ViewGroup parent) {
            CheckedTextView text= (CheckedTextView) convertView;

            if (text== null) {
                text = (CheckedTextView) LayoutInflater.from(getContext()).inflate(android.R.layout.simple_spinner_dropdown_item,  parent, false);
            }

            text.setText(getItem(position).toString(con));

            return text;
        }

        @Override
        public View getDropDownView(int position, View convertView, ViewGroup parent) {
            CheckedTextView text = (CheckedTextView) convertView;

            if (text == null) {
                text = (CheckedTextView) LayoutInflater.from(getContext()).inflate(android.R.layout.simple_spinner_dropdown_item,  parent, false);
            }

            text.setText(getItem(position).toString(con));

            return text;
        }
    }




}

File: src/main/java/com/liamfarrell/android/snapbattle/mvvm_ui/create_battle/ChooseVotingFragment.kt
package com.liamfarrell.android.snapbattle.mvvm_ui.create_battle

import android.content.Context
import android.os.Bundle
import android.view.*
import android.widget.RadioButton
import androidx.appcompat.app.AppCompatActivity
import androidx.core.os.bundleOf
import androidx.fragment.app.Fragment
import androidx.navigation.fragment.findNavController
import com.liamfarrell.android.snapbattle.R
import com.liamfarrell.android.snapbattle.adapters.VotingLengthAdapter
import com.liamfarrell.android.snapbattle.databinding.FragmentChooseVotingBinding
import kotlinx.android.synthetic.main.fragment_choose_voting.*

class ChooseVotingFragment : Fragment(), View.OnClickListener{

    private var chosenVotingChoice: VotingChoice? = VotingChoice.NONE


    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceBundle: Bundle?): View? {
        val binding = FragmentChooseVotingBinding.inflate(inflater, container, false)
        binding.lifecycleOwner = viewLifecycleOwner
        binding.noVotingButton.setOnClickListener(this)
        binding.publicVotingButton.setOnClickListener(this)
        binding.mutualFacebookFriendsButton.setOnClickListener(this)
        binding.votingTypeRadioGroup.check(R.id.noVotingButton)

        // Apply the adapter to the spinner
        binding.votingLengthSpinner.adapter = VotingLengthAdapter(requireContext())
        binding.votingLengthSpinner.isEnabled = false
        binding.votingLengthSpinner.setSelection(0)
        setToolbar(binding.includeToolbar.toolbar)
        setHasOptionsMenu(true)
        return binding.root
    }

    private fun setToolbar(toolbar : androidx.appcompat.widget.Toolbar){
        (activity as AppCompatActivity).setSupportActionBar(toolbar)
        (activity as AppCompatActivity).supportActionBar?.title = "Choose Voting";
        (activity as AppCompatActivity).supportActionBar?.setDisplayHomeAsUpEnabled(false);
    }

    override fun onCreateOptionsMenu(menu: Menu, inflater : MenuInflater) {
        super.onCreateOptionsMenu(menu, inflater)
        inflater.inflate(R.menu.menu_create_battle, menu);
    }

    override fun onOptionsItemSelected(item : MenuItem) : Boolean {
        val id = item.itemId
        if (id == R.id.action_next) {
            val bundle = bundleOf(("votingLength" to votingLengthSpinner.selectedItem.toString()),
                    ("votingType" to chosenVotingChoice.toString()))
            bundle.putAll(arguments)
            findNavController().navigate(R.id.action_chooseVotingFragment_to_verifyBattleFragment, bundle)
        }
        return false
    }

    override fun onClick(view: View) {
        // determine which voting option is checked and update the member variables
        val checked = (view as RadioButton).isChecked
        // Check which radio button was clicked
        when (view.getId()) {
            R.id.noVotingButton -> if (checked) {
                votingLengthSpinner.isEnabled = false
                chosenVotingChoice = VotingChoice.NONE
            }
            R.id.publicVotingButton -> if (checked) {
                votingLengthSpinner.isEnabled = true
                chosenVotingChoice = VotingChoice.PUBLIC
            }
            R.id.mutualFacebookFriendsButton -> if (checked) {
                votingLengthSpinner.isEnabled = true
                chosenVotingChoice = VotingChoice.MUTUAL_FACEBOOK
            }
        }
    }


    enum class VotingChoice {

        MUTUAL_FACEBOOK {
            override fun toLongStyleString(context: Context): String {
                return context.resources.getText(R.string.mutual_facebook_friends_long).toString()
            }

            override fun toString(context: Context): String {
                return context.resources.getText(R.string.mutual_facebook_friends).toString()
            }
        },
        PUBLIC {
            override fun toLongStyleString(context: Context): String {
                return context.resources.getText(R.string.public_voting_long).toString()
            }

            override fun toString(context: Context): String {
                return context.resources.getText(R.string.public_voting).toString()
            }
        },
        SELECTED {
            override fun toLongStyleString(context: Context): String {
                return context.resources.getText(R.string.selected_judges).toString()
            }

            override fun toString(context: Context): String {
                return context.resources.getText(R.string.selected_judges).toString()
            }
        },
        NONE {
            override fun toLongStyleString(context: Context): String {
                return context.resources.getText(R.string.no_voting_used).toString()
            }

            override fun toString(context: Context): String {
                return context.resources.getText(R.string.no_voting_used).toString()
            }
        };

        abstract fun toString(context: Context): String
        abstract fun toLongStyleString(context: Context): String
    }


    enum class VotingLength {
        TWENTY_FOUR_HOURS {
            override fun toString(context: Context): String {
                return context.resources.getStringArray(R.array.VotingLengths)[0]
            }
        },
        THREE_DAYS {
            override fun toString(context: Context): String {
                return context.resources.getStringArray(R.array.VotingLengths)[1]
            }
        },
        ONE_WEEK {
            override fun toString(context: Context): String {
                return context.resources.getStringArray(R.array.VotingLengths)[2]
            }
        },
        ONE_MONTH {
            override fun toString(context: Context): String {
                return context.resources.getStringArray(R.array.VotingLengths)[3]
            }
        };

        abstract fun toString(context: Context): String
    }



}